src/com/sun/ts/lib/util/TSNamingContext.java

```diff
@@ -53,7 +51,7 @@ public TSNamingContext(Properties pp) throws Exception {
    */
   public Object lookup(String s, Class c) throws Exception {
     Object o = lookup(s);
-    return c == null ? o : PortableRemoteObject.narrow(o, c);
+    return c == null ? o : o;
```

Comment:
If `c` is not null, should we throw a ClassCastException failure if the returned object is not an instance of `Class c`?